In [1]:
from google.colab import drive
drive.mount('/content/drive/')

Drive already mounted at /content/drive/; to attempt to forcibly remount, call drive.mount("/content/drive/", force_remount=True).


In [ ]:
!pip install -r /content/drive/MyDrive/project/semantic-segmentation-roads/eomt/requirements.txt

  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 52.0/52.0 kB 3.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.2/42.2 kB 4.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.0/62.0 kB 6.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.2/50.2 kB 5.3 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.2/42.2 kB 4.4 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.0/61.0 kB 6.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 219.4/219.4 kB 14.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.6/8.6 MB 111.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.4/2.4 MB 108.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 819.0/819.0 kB 61.0 MB

In [3]:
%cd /content/drive/MyDrive/project/semantic-segmentation-roads/eval

/content/drive/.shortcut-targets-by-id/1kGi4cSNJjM14ClVvPXJ9VY70JDkofHGn/project/semantic-segmentation-roads/eval


In [ ]:
! python evalAnomalyEOMT_scaling.py --base_dir "/content/drive/MyDrive/project/Anomaly_Validation_Datasets/Validation_Dataset" --model 'cityscapes' --save_inf

2026-05-28 11:48:23.745753: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
/usr/local/lib/python3.12/dist-packages/lightning/pytorch/utilities/parsing.py:209: Attribute 'network' is an instance of `nn.Module` and is already saved during checkpointing. It is recommended to ignore them using `self.save_hyperparameters(ignore=['network'])`.
Caricamento pesi da: /content/drive/MyDrive/project/models_weights/eomt_cityscapes.bin
Missing keys: []
Unexpected keys: []
Modello EoMT caricato con successo.

 Inizio valutazione Dataset: FS_LostFound_full
  [100/100] Elaborando 91.png...
 -> MSP        | AUPRC: 16.39% | FPR@TPR95: 13.01%
 -> MAXLOGIT   | AUPRC: 16.38% | FPR@TPR95: 12.78%
 -> MAXENTROPY | AUPRC: 19.08% | FPR@TPR95: 12.83%
 -> RBA        | AUPR

In [2]:
!pip install ood-metrics

In [4]:
!python temperature.py


 OTTIMIZZAZIONE TEMPERATURA PER MODELLO: CITYSCAPES

--- Dataset: RoadAnomaly ---
  T = 0.25 | MSP -> AUPRC: 72.07% | FPR@TPR95: 16.26%
  T = 0.50 | MSP -> AUPRC: 72.26% | FPR@TPR95: 15.75%
  T = 0.75 | MSP -> AUPRC: 72.38% | FPR@TPR95: 15.58%
  T = 1.00 | MSP -> AUPRC: 72.46% | FPR@TPR95: 15.53%
  T = 1.10 | MSP -> AUPRC: 72.49% | FPR@TPR95: 15.52%
  T = 1.25 | MSP -> AUPRC: 72.52% | FPR@TPR95: 15.51%
  T = 1.50 | MSP -> AUPRC: 72.58% | FPR@TPR95: 15.49%
  T = 2.00 | MSP -> AUPRC: 72.66% | FPR@TPR95: 15.48%
  T = 3.00 | MSP -> AUPRC: 72.78% | FPR@TPR95: 15.46%
  T = 5.00 | MSP -> AUPRC: 72.96% | FPR@TPR95: 15.45%

--- Dataset: RoadAnomaly21 ---
  T = 0.25 | MSP -> AUPRC: 67.79% | FPR@TPR95: 30.40%
  T = 0.50 | MSP -> AUPRC: 67.94% | FPR@TPR95: 30.42%
  T = 0.75 | MSP -> AUPRC: 68.03% | FPR@TPR95: 30.42%
  T = 1.00 | MSP -> AUPRC: 68.10% | FPR@TPR95: 30.42%
  T = 1.10 | MSP -> AUPRC: 68.12% | FPR@TPR95: 30.42%
  T = 1.25 | MSP -> AUPRC: 68.15% | FPR@TPR95: 30.42%
  T = 1.50 | MSP -> A


=======================================================
 OTTIMIZZAZIONE TEMPERATURA PER MODELLO: CITYSCAPES
=======================================================

--- Dataset: FS_LostFound_full ---
  T = 0.25 | MSP -> AUPRC: 16.36% | FPR@TPR95: 13.03%
  T = 0.50 | MSP -> AUPRC: 16.38% | FPR@TPR95: 13.02%
  T = 0.75 | MSP -> AUPRC: 16.39% | FPR@TPR95: 13.01%
  T = 1.00 | MSP -> AUPRC: 16.39% | FPR@TPR95: 13.01%
  T = 1.10 | MSP -> AUPRC: 16.40% | FPR@TPR95: 13.01%
  T = 1.25 | MSP -> AUPRC: 16.40% | FPR@TPR95: 13.01%
  T = 1.50 | MSP -> AUPRC: 16.40% | FPR@TPR95: 13.01%
  T = 2.00 | MSP -> AUPRC: 16.41% | FPR@TPR95: 13.01%
  T = 3.00 | MSP -> AUPRC: 16.43% | FPR@TPR95: 13.01%

In [ ]:
!python temperature.py


 OTTIMIZZAZIONE TEMPERATURA PER MODELLO: CITYSCAPES

--- Dataset: FS_LostFound_full ---
  T = 0.25 | MSP -> AUPRC: 16.36% | FPR@TPR95: 13.03%
  T = 0.50 | MSP -> AUPRC: 16.38% | FPR@TPR95: 13.02%
  T = 0.75 | MSP -> AUPRC: 16.39% | FPR@TPR95: 13.01%
  T = 1.00 | MSP -> AUPRC: 16.39% | FPR@TPR95: 13.01%
  T = 1.10 | MSP -> AUPRC: 16.40% | FPR@TPR95: 13.01%
  T = 1.25 | MSP -> AUPRC: 16.40% | FPR@TPR95: 13.01%
  T = 1.50 | MSP -> AUPRC: 16.40% | FPR@TPR95: 13.01%
  T = 2.00 | MSP -> AUPRC: 16.41% | FPR@TPR95: 13.01%
  T = 3.00 | MSP -> AUPRC: 16.43% | FPR@TPR95: 13.01%


In [ ]:
!python temperature.py


 OTTIMIZZAZIONE TEMPERATURA PER MODELLO: CITYSCAPES

--- Dataset: FS_LostFound_full ---
  T = 5.00 | MSP -> AUPRC: 16.45% | FPR@TPR95: 13.01%
